### ISSUE TITLE
Move the unstable-api dump tool to the main rust repo

### ISSUE THREAD

**Issue by albertlarsan68** [rust-lang/libs-team#161]
This will allow to use the json rustdoc output, thus having the macros expanded for free, and there is already code for finding this in tidy and unstable-book-gen.

**Comment by m-ou-se** [comment]
Yeah it'd be cool to have a unstable-api tool that's based on rustc/rustdoc, rather than syn. :)
